lib/AST/LayoutConstraint.cpp

```diff
@@ -0,0 +1,61 @@
+//===- LayoutConstraint.cpp - Layout constraints types and APIs -*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements APIs for layout constraints.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/AST/Decl.h"
+#include "swift/AST/Types.h"
+#include "swift/AST/TypeRepr.h"
+#include "llvm/ADT/StringSwitch.h"
+#include "cctype"
+
+namespace swift {
+
+LayoutConstraintInfo getLayoutConstraintInfo(StringRef ID) {
+  if (ID == "_Trivial")
```

Comment:
May be instead I should define these identifiers by means of x-macros as static members of LayoutConstraint, so that it is possible to access them anywhere using`LayoutConstraint::Id_Trivial`, etc?